pulpcore/cli/rpm/context.py

```diff
@@ -33,6 +35,21 @@ def refresh(self, href: str) -> Any:
         )
 
 
+class PulpRpmCompsXmlContext(PulpEntityContext):
+    UPLOAD_COMPS_ID: ClassVar[str] = "rpm_comps_upload"
+
+    def upload_comps(
+        self, file: IO[bytes], repo_href: Optional[str], replace: Optional[bool]
+    ) -> Any:
+        click.echo(_("Uploading file {filename}").format(filename=file.name), err=True)
+        file.seek(0)
+        return self.pulp_ctx.call(
+            self.UPLOAD_COMPS_ID,
+            uploads={"file": file.read()},
```

Comment:
Are we quite confident, these never need to go through the uploads api for being too big?
(From the CLI view, i kind of think, there should only have been a single way to upload data into pulp, but at least every other serializer that accepts files, should also accept uploads or artifacts.)

Reply:
comps.xml isn't recognized by pulp_rpm as an entity itself - it's a convenient way of getting the PackageCatalog/Environment/Langpack/Groups uploaded, but there is no 'comps.xml' file anywhere. As far as size goes, the largest comps.xml I know of is Fedora-Updates; F34 unzipped is 1.6Mb. That's def Large; you're right, maybe I need to rethink "it's just a file".

Reply:
The chunked-upload-api doesn't play very well with either "convenience API" or "not-an-artifact". Given that the largest comps.xml I know if is 1.7Mb, comps.xml isn't its own artifact, and the pulp_rpm create calls all have --file= parameters, I think this is the way to go.

If at some point we decide to, say, add a comps-Artifact and extends the POST /comps/ call  to include an --artifact option, we can extend the CLI to handle it using chunked uploads. But for the purposes of filling the RFE, I think this is sufficient, and consistent.